chain/rust/src/utils.rs

```diff
@@ -569,6 +569,302 @@ impl Deserialize for NetworkId {
     }
 }
 
+// Represents the cddl: #6.258([+ T]) / [* T]
+// it DOES NOT and CAN NOT have any encoding detials per element!
+// so you can NOT use it on any primitives so must be serializable directly
+#[derive(Debug, Clone)]
+pub struct NonemptySet<T> {
```

Comment:
does this have to be capitalized in this weird way for some reason?

Reply:
it's just how it is in the conway.cddl